src/compiler.ts

```diff
@@ -4713,18 +4713,17 @@ export class Compiler extends DiagnosticEmitter {
           }
           case TypeKind.ISIZE:
           case TypeKind.USIZE: {
-            let type = this.options.isWasm64
-              ? (this.currentType.is(TypeFlags.SIGNED) ? Type.i64 : Type.u64)
-              : (this.currentType.is(TypeFlags.SIGNED) ? Type.i32 : Type.u32);
+            let isWasm64 = this.options.isWasm64;
+            let type = isWasm64 ? Type.isize64 : Type.isize32;
```

Comment:
Isn't `this.currentType` the type this branch wants already? Looks like this can just use `this.currentType` instead, and skip the `leftExpr = this.convertExpression` below.